include/unifex/variant_sender.hpp

```diff
@@ -71,22 +71,27 @@ struct _op<Ops...>::type {
   variant<manual_lifetime<Ops>...> variantOp_;
 };
 
-template <_block::_enum First, _block::_enum... Rest>
+template <typename Sender, typename... Rest>
 struct max_blocking_kind {
-  constexpr auto operator()() noexcept { return First; }
+  constexpr auto operator()() noexcept { return cblocking<Sender>(); }
 };
 
-template <_block::_enum First, _block::_enum Second, _block::_enum... Rest>
+template <typename First, typename Second, typename... Rest>
 struct max_blocking_kind<First, Second, Rest...> {
   constexpr auto operator()() noexcept {
-    if constexpr (First == Second) {
+    constexpr blocking_kind first = cblocking<First>();
+    constexpr blocking_kind second = cblocking<Second>();
+
+    if constexpr (first == second) {
+      return max_blocking_kind<First, Rest...>{}();
+    } else if constexpr (
+        first == blocking_kind::always &&
+        second == blocking_kind::always_inline) {
       return max_blocking_kind<First, Rest...>{}();
     } else if constexpr (
-        (First == blocking_kind::always &&
-        Second == blocking_kind::always_inline) ||
-        (Second == blocking_kind::always &&
-        First == blocking_kind::always_inline)) {
-      return max_blocking_kind<blocking_kind::always, Rest...>{}();
+        first == blocking_kind::always_inline &&
+        second == blocking_kind::always) {
+      return max_blocking_kind<Second, Rest...>{}();
     } else {
       return blocking_kind::maybe;
```

Comment:
Do you also want to handle the case where First and Second are both 'never' to map to max_blocking_kind<never, Rest...>?